### PR TITLE
Updating readme to include guide on integrating with docker agent

### PR DESCRIPTION
**Pull request by chenbishop**
### What does this PR do?

Adding steps for integrating with docker agent

### Motivation

User feature request.

### Review checklist

- [ ] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [ ] Feature or bugfix has tests
- [ ] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Comment by drichards-87** [comment]
Created DOCS-5432 for the Docs team review.

**Review by estherk15 — COMMENTED** [review]
Hi @chenbishop thanks for contributing to the our docs. Left a few notes on your requested changes, please let me know if you have any questions!

**Comment by chenbishop** [comment]
@estherk15 thank you very much for your feedback, i have made all the recommended changes.

File: twingate/README.md [hunk 1]
```diff
@@ -33,8 +33,61 @@ This integration allows organizations to monitor a user's resource access activi
...
    ```
 3. Add the `dd-agent` user to the `systemd-journal` group by using `usermod -a -G systemd-journal dd-agent`.
-4. Restart the Datadog Agent by running `service datadog-agent restart`.
-5. Confirm that the Twingate Analytic log appears in the [Log Explorer][10].
+4. Restart the Datadog Agent by running `service datadog-agent restart`
+5. Confirm that the Twingate Analytic log appears in the [Log Explorer][10]
+
+
+#### Docker Connector
+##### (Host Agent) Set up Datadog Docker integration
```
Comment (estherk15) [inline twingate/README.md:41]:
```suggestion
##### Set up Datadog Docker integration for the Host Agent
```

File: twingate/README.md [hunk 2]
```diff
@@ -33,8 +33,61 @@ This integration allows organizations to monitor a user's resource access activi
...
+- Adding the following lines to `datadog.yaml`:
```
Comment (estherk15) [inline twingate/README.md:42]:
```suggestion
Add the following lines to the `datadog.yaml` configuration file:
```

File: twingate/README.md [hunk 3]
```diff
@@ -33,8 +33,61 @@ This integration allows organizations to monitor a user's resource access activi
...
+```
```
Comment (estherk15) [inline twingate/README.md:43]:
```suggestion
```yaml
```

File: twingate/README.md [hunk 4]
```diff
@@ -33,8 +33,61 @@ This integration allows organizations to monitor a user's resource access activi
...
+polling: true
+logs_config:
+container_collect_all: true
+container_exclude: ["image:.*"]
+container_include: ["image:twingate/connector"]
+```
+- Add the `dd-agent` user to the `docker` group by using `usermod -a -G docker dd-agent`
+- Restart the Datadog Agent by running `service datadog-agent restart`
+
+##### (Container Agent) Set up Datadog Docker integration
```
Comment (estherk15) [inline twingate/README.md:58]:
```suggestion
##### Set up Datadog Docker integration for the Container Agent
```

File: twingate/README.md [hunk 5]
```diff
@@ -33,8 +33,61 @@ This integration allows organizations to monitor a user's resource access activi
...
+- name: docker
+config_providers:
+- name: docker
+polling: true
+logs_config:
+container_collect_all: true
+container_exclude: ["image:.*"]
+container_include: ["image:twingate/connector"]
+```
+- Add the `dd-agent` user to the `docker` group by using `usermod -a -G docker dd-agent`
```
Comment (estherk15) [inline twingate/README.md:55]:
```suggestion
- Add the `dd-agent` user to the `docker` group by using `usermod -a -G docker dd-agent`.
```

File: twingate/README.md [hunk 6]
```diff
@@ -33,8 +33,61 @@ This integration allows organizations to monitor a user's resource access activi
...
+- Restart the Datadog Agent by running `service datadog-agent restart`
```
Comment (estherk15) [inline twingate/README.md:56]:
```suggestion
- Restart the Datadog Agent by running `service datadog-agent restart`.
```

File: twingate/README.md [hunk 7]
```diff
@@ -33,8 +33,61 @@ This integration allows organizations to monitor a user's resource access activi
           pattern: "ANALYTICS "
    ```
 3. Add the `dd-agent` user to the `systemd-journal` group by using `usermod -a -G systemd-journal dd-agent`.
-4. Restart the Datadog Agent by running `service datadog-agent restart`.
-5. Confirm that the Twingate Analytic log appears in the [Log Explorer][10].
+4. Restart the Datadog Agent by running `service datadog-agent restart`
```
Comment (estherk15) [inline twingate/README.md:36]:
```suggestion
4. Restart the Datadog Agent by running `service datadog-agent restart`.
```

File: twingate/README.md [hunk 8]
```diff
@@ -33,8 +33,61 @@ This integration allows organizations to monitor a user's resource access activi
...
+5. Confirm that the Twingate Analytic log appears in the [Log Explorer][10]
```
Comment (estherk15) [inline twingate/README.md:37]:
```suggestion
5. Confirm that the Twingate Analytic log appears in the [Log Explorer][10].
```

File: twingate/README.md [hunk 9]
```diff
@@ -33,8 +33,61 @@ This integration allows organizations to monitor a user's resource access activi
...
+container_collect_all: true
+container_exclude: ["image:.*"]
+container_include: ["image:twingate/connector"]
+```
+- Add the `dd-agent` user to the `docker` group by using `usermod -a -G docker dd-agent`
+- Restart the Datadog Agent by running `service datadog-agent restart`
+
+##### (Container Agent) Set up Datadog Docker integration
+- add additional parameters `-e DD_CONTAINER_EXCLUDE="image:.*"` and `-e DD_CONTAINER_INCLUDE="image:twingate/connector"` in the docker run command
+```
```
Comment (estherk15) [inline twingate/README.md:60]:
```suggestion
```shell
```

File: twingate/README.md [hunk 10]
```diff
@@ -33,8 +33,61 @@ This integration allows organizations to monitor a user's resource access activi
...
+           -v /var/run/docker.sock:/var/run/docker.sock:ro \
+           -v /var/lib/docker/containers:/var/lib/docker/containers:ro \
+           -v /proc/:/host/proc/:ro \
+           -v /opt/datadog-agent/run:/opt/datadog-agent/run:rw \
+           -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
+           gcr.io/datadoghq/agent:latest
+```
+
+##### Set up Twingate Connector with additional docker parameters
+Adding additional label `com.datadoghq.ad.logs` to the Twingate Connector docker run command:
```
Comment (estherk15) [inline twingate/README.md:78]:
```suggestion
Add the label `com.datadoghq.ad.logs` to the Twingate Connector docker run command:
```

File: twingate/README.md [hunk 11]
```diff
@@ -33,8 +33,61 @@ This integration allows organizations to monitor a user's resource access activi
...
+```
```
Comment (estherk15) [inline twingate/README.md:79]:
```suggestion
```shell
```

File: twingate/README.md [hunk 12]
```diff
@@ -33,8 +33,61 @@ This integration allows organizations to monitor a user's resource access activi
...
+  -l "com.datadoghq.ad.logs"='[{"service":"Twingate Connection","source":"Twingate","log_processing_rules":[{"type":"include_at_match","name":"analytics","pattern":"ANALYTICS"},{"type":"mask_sequences","name":"remove_analytics","replace_placeholder":"","pattern":"ANALYTICS "}]}]' \
+  --env TENANT_URL="https://xxx.twingate.com" \
+  --env ACCESS_TOKEN="xxx" \
+  --env REFRESH_TOKEN="xxx" \
+  --env TWINGATE_LABEL_HOSTNAME="`hostname`" \
+  --name "twingate-golden-seal" \
+  --restart=unless-stopped \
+  $(docker run --help | grep -- --pull >/dev/null && echo "--pull=always") twingate/connector:1
+```
+Note: The Twingate Connector container needs to be recreated to add the additional label
```
Comment (estherk15) [inline twingate/README.md:90]:
```suggestion
**Note**: The Twingate Connector container needs to be recreated to add the new label 
```

File: twingate/README.md [hunk 13]
```diff
@@ -55,3 +108,10 @@ Need help? Contact [Twingate Support][2].
 [10]: https://app.datadoghq.com/logs
 [11]: https://docs.datadoghq.com/getting_started/agent/
 [12]: https://app.datadoghq.com/dashboard/lists
+[13]: https://docs.datadoghq.com/containers/docker/log/?tab=containerinstallation#installation
+[14]: https://docs.datadoghq.com/agent/guide/autodiscovery-management/?tab=agent_
+[15]: https://docs.datadoghq.com/containers/docker/log/?tab=containerinstallation#log-integrations
+[16]: https://gist.github.com/chenbishop/3f1b9833f274716949a40af385e2f5c2
```
Comment (estherk15) [inline twingate/README.md:114]:
```suggestion
```

File: twingate/README.md [hunk 14]
```diff
@@ -55,3 +108,10 @@ Need help? Contact [Twingate Support][2].
...
+[17]: https://gist.github.com/chenbishop/7df60e074e7e856e17d867b31f6e95a1
```
Comment (estherk15) [inline twingate/README.md:115]:
```suggestion
```

File: twingate/README.md [hunk 15]
```diff
@@ -55,3 +108,10 @@ Need help? Contact [Twingate Support][2].
...
+[18]: https://gist.github.com/chenbishop/3be2bcc90d7d77288a843de469f0b3d0
```
Comment (estherk15) [inline twingate/README.md:116]:
```suggestion
```

File: twingate/README.md [hunk 16]
```diff
@@ -55,3 +108,10 @@ Need help? Contact [Twingate Support][2].
...
+[19]: https://gist.github.com/chenbishop/a5405cc00bc1ac50fdbd86ef4ad23599
```
Comment (estherk15) [inline twingate/README.md:117]:
```suggestion
```

File: twingate/README.md [hunk 17]
```diff
@@ -12,7 +12,7 @@ This integration allows organizations to monitor a user's resource access activi
 2. You must deploy the Twingate Connector. To enable real-time connection logs, see the [Twingate documentation][3].
 
 ### Configure the Datadog Agent
-#### Systemd
+#### Systemd Connector
 1. Set up [Datadog journald integration][5].
 2. Replace `journald.d/conf.yaml` with the following configuration:
    ```
```
Comment (estherk15) [inline twingate/README.md:18]:
```suggestion
   ```yaml
```

**Review by estherk15 — CHANGES_REQUESTED** [review]
Hi @chenbishop thanks for making those changes! A few more formatting changes and cleanup for the extra links. Please let me know if you have any questions!

**Review by estherk15 — CHANGES_REQUESTED** [review]
Hi @chenbishop thanks for making those changes! A few more formatting changes and cleanup for the extra links. Please let me know if you have any questions!

**Comment by chenbishop** [comment]
@estherk15 thank you very much for suggested changes, i have committed all the suggested changes.

**Review by mgarabed — APPROVED** [review]

**Pull request merged**